gott.go

```diff
@@ -50,57 +50,44 @@ func (c config) Flatten() map[string]string {
 	return results
 }
 
-// return a copy of self with path promoted to top level data
-func (c config) Promote(path []string) config {
-	result := config{}
-	mergo.Merge(&result, c)
-
-	if err := mergo.Merge(&result, c.Narrow(path)); err != nil {
-		panic(err)
-	}
-
-	return result
-}
-
-func (c config) Narrow(path []string) config {
-	// "copy"
-	result := config{}
-	mergo.Merge(&result, c)
-
-	var dig map[string]interface{} = c
-	for _, key := range path {
-		dig = dig[key].(map[string]interface{})
-	}
-
-	return dig
-}
-
-func (c config) Render(template_text string) string {
+func (c config) Render(template_text string) (string, error) {
 	t := template.Must(template.New("base").Funcs(sprig.TxtFuncMap()).Parse(template_text))
 	result := new(bytes.Buffer)
 	err := t.Execute(result, c)
-
 	if err != nil {
-		panic(err)
+		return "", err
 	}
-	return result.String()
+	return result.String(), nil
 }
 
 // Have a config render itself
-func realizeConfig(m map[string]interface{}, config config, path []string) map[string]interface{} {
-	for k, v := range m {
-		switch v := v.(type) {
-		// case []interface{}:
-			// for i, v := range v {
-			// 	// m[k][i] = walk(v, config, path)
-			// }
+func realizeConfig(subConfig config, rootConfig config) error {
+	for key, value := range subConfig {
+		switch value := value.(type) {
 		case map[string]interface{}:
-			m[k] = realizeConfig(v, config, append(path, k))
+			err := realizeConfig(value, rootConfig)
+			if err != nil {
+				return err
+			}
+
 		case string:
-			m[k] = config.Promote(path).Render(v)
+			realizedValue, err := subConfig.Render(value)
+			if err == nil {
+				subConfig[key] = realizedValue
+				continue
+			}
+
+			realizedValue, err = rootConfig.Render(value)
+			if err == nil {
+				subConfig[key] = realizedValue
+				continue
+			}
```

Comment:
the bug here is that we may not have realized the key that `value` is pointing towards, so we can get bugs like:

```
~/src/gott$ rm -rf ~/.cache/gott && go run gott.go -t font.toml -o toml
[font]
config = 'Charter 16'
family = 'Charter'
size = 16

[loud]
font-config-loud = '{{.FAMILY}} {{.SIZE}}'
```